platforms/android/src/jp/wizcorp/phonegap/plugin/WizAssets/WizAssetsPlugin.java

```diff
@@ -317,80 +330,147 @@ public AsyncDownload(String url, String uri, String filePath, CallbackContext ca
+                    if (credentials != null) {
+                        // Add Basic Authentication header
+                        httpRequest.setHeader("Authorization", "Basic " + Base64.encodeToString(credentials.getBytes(), Base64.NO_WRAP));
+                    }
+                } catch (MalformedURLException e) {
+                    this.callbackContext.error(createDownloadFileError(INVALID_URL_ERROR));
+                    return null;
+                } catch (URISyntaxException e) {
+                    this.callbackContext.error(createDownloadFileError(INVALID_URL_ERROR));
+                    return null;
+                }
 
+                HttpResponse response;
                 HttpClient httpclient = new DefaultHttpClient();
-
-                // Credential check
-                String credentials = url.getUserInfo();
-                if (credentials != null) {
-                    // Add Basic Authentication header
-                    httpRequest.setHeader("Authorization", "Basic " + Base64.encodeToString(credentials.getBytes(), Base64.NO_WRAP));
+                try {
+	                response = httpclient.execute(httpRequest);
+	                if (response == null) {
+                        this.callbackContext.error(createDownloadFileError(CONNECTIVITY_ERROR));
+                        return null;
+	                }
+	                StatusLine statusLine = response.getStatusLine();
+	                if (statusLine == null) {
+                        this.callbackContext.error(createDownloadFileError(HTTP_REQUEST_ERROR, "No status available"));
+                        return null;
+                    }
+                    int statusCode = statusLine.getStatusCode();
+                    if (statusCode < 200 || statusCode > 299) {
+                        this.callbackContext.error(createDownloadFileError(HTTP_REQUEST_ERROR, statusCode));
+                        return null;
+                    }
+                } catch (ClientProtocolException e) {
+                    this.callbackContext.error(createDownloadFileError(CONNECTIVITY_ERROR));
+                    return null;
+                } catch (IOException e) {
+                    this.callbackContext.error(createDownloadFileError(CONNECTIVITY_ERROR));
+                    return null;
                 }
 
-                HttpResponse response = httpclient.execute(httpRequest);
                 HttpEntity entity = response.getEntity();
 
                 InputStream is;
 
                 Header contentHeader = entity.getContentEncoding();
-                if (contentHeader != null) {
-                    if (contentHeader.getValue().contains("gzip")) {
-                        Log.d(TAG, "GGGGGGGGGZIIIIIPPPPPED!");
+                try {
+                    String contentHeaderValue = contentHeader != null ? contentHeader.getValue() : null;
+                    if (contentHeaderValue != null && contentHeaderValue.contains("gzip")) {
```

Comment:
If that's ok with you, I won't fix it now, I created an issue to keep track of it. See #17.